src/prices/coingecko.ts

```diff
@@ -0,0 +1,53 @@
+import BigNumber from 'bignumber.js'
+import { logger } from '../logger'
+import fetch from 'cross-fetch'
+
+export const BASE_URL =
+  'https://api.coingecko.com/api/v3/simple/token_price/celo'
+
+const USD = 'usd'
+
+/**
+ * Returns token prices compared to the base token.
+ */
+export async function fetchPrices(tokenAddresses: string[], baseToken: string) {
+  try {
+    const url = `${BASE_URL}?contract_addresses=${tokenAddresses.join(
+      ',',
+    )}&vs_currencies=usd`
+    logger.debug(`Fetching prices with ${url}`)
+    const response = await fetch(url)
+    const jsonResponse = await response.json()
+    logger.debug(`Response from coingecko: ${JSON.stringify(jsonResponse)}`)
+    const prices: Record<string, BigNumber> = {}
+
+    const baseTokenUsdPrice = jsonResponse[baseToken]
+      ? new BigNumber(jsonResponse[baseToken][USD])
+      : null
+    if (!baseTokenUsdPrice) {
+      logger.error({
+        type: 'MISSING_BASE_TOKEN_PRICE_FROM_COINGECKO',
+      })
+      return {}
+    }
+
+    for (const tokenAddress of tokenAddresses) {
+      if (jsonResponse[tokenAddress] && jsonResponse[tokenAddress][USD]) {
+        const usdPrice = jsonResponse[tokenAddress][USD]
+        prices[tokenAddress] = new BigNumber(usdPrice).dividedBy(
+          baseTokenUsdPrice,
+        )
+      }
+    }
+
+    return prices
+  } catch (e) {
+    const error = e as Error
+    logger.error({
+      error,
+      errorMessage: error.message,
+      type: 'ERROR_FETCHING_PRICES_FROM_COINGECKO',
+    })
```

Comment:
The default [error serializer](https://github.com/trentm/node-bunyan/blob/5c2258ecb1d33ba34bd7fbd6167e33023dc06e40/lib/bunyan.js#L1141) adds more info (stack, name, code) when captured like this:
```suggestion
  } catch (err) {
    logger.error({
      err,
      errorMessage: error.message,
      type: 'ERROR_FETCHING_PRICES_FROM_COINGECKO',
    })
```